src/requestforattestation/RequestForAttestation.ts

```diff
@@ -108,21 +150,40 @@ export default class RequestForAttestation implements IRequestForAttestation {
   public constructor(
     claim: IClaim,
     legitimations: AttestedClaim[],
-    identity: Identity,
+    claimOwner: NonceHash,
+    ctypeHash: NonceHash,
+    claimHashTree: object,
+    identity?: Identity,
+    claimerSignature?: string,
     delegationId?: IDelegationBaseNode['id']
   ) {
-    if (claim.owner !== identity.address) {
-      throw Error('Claim owner is not identity')
-    }
-    this.claim = claim
-    this.claimOwner = generateHash(this.claim.owner)
-    this.ctypeHash = generateHash(this.claim.cType)
-    this.legitimations = legitimations
-    this.delegationId = delegationId
+    if (identity !== undefined) {
```

Comment:
Also check for `null`. 
Because if the user wants to pass `claimerSignature` but not `identity` nor `delegationId`: 
* They might use `undefined`s. In that case, no problem.
* But it might also feel reasonable to call ```new RequestForAttestation(claim, legitimations, claimOwner, ctypeHash, claimHashTree, null, claimerSignature , null)```. We can't really make assumptions on what they would use.

General note / suggestion for the future:

Right now, there a few small dev experience issues with the constructor (although it's much better than before!!):
* 1: The issue mentioned above
* 2: if the user calls ```new RequestForAttestation(claim, legitimations, claimOwner, ctypeHash, claimHashTree, identity, delegationId)```, we'll assign `delegationId` as a `signature`. This would be the user's mistake, but it doesn't mean we can't protect them.
* 3: We enforce passing some `undefined`s or `null`s even when the user doesn't need them.

Solution to these 3 issues (for the future / in another ticket maybe?):
We might want to just pass the params as a config object (or interface, probably better): 
```constructor({claim: claim, legitimations: legitimations...```

@tjwelde any thoughts?

Reply:
Timos input is needed for that.

Reply:
For the `RequestForAttestation` Constructor we pass arguments that do not end up in the final instantiated Object, the Identity the final RequestForAttestation is signed with.